### PR TITLE
Update code tidy, add ability to enter password

### PR DESCRIPTION
**Pull request by EventReplay**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,61 +25,61 @@ instance.prototype.updateConfig = function(config) {
...
+		self.status(self.STATUS_OK);
+		debug("logged in");
 	}
 	// Heatbeat to keep connection alive
+	function heartbeat() {
+		self.login = false;
+		self.status(self.STATUS_WARNING,'Checking Connection');
+		self.socket.write("N"+ "\n"); // should respond with Switcher part number eg: "60-882-01" = DXP 88 HDMI
+		debug("Checking Connection");
+		}
```
Comment (haakonnessjoen) [inline index.js]: Not to be nitpicky, but the right curly bracket here is indented wrong. Also, you could change "N"+"\n", to simply "N\n" ;)

File: index.js [hunk 2]
```diff
@@ -25,61 +25,61 @@ instance.prototype.updateConfig = function(config) {
...
 	if (self.login === false && data.match("Extron Electronics")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write("I"+ "\n");
+		self.socket.write("I"+ "\n"); // Matrix information request
 	}
 
 	if (self.login === false && data.match("Password:")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write(""+ "\n");
+		self.socket.write("\r" +self.config.password+ "\r"); // Enter Password Set
```
Comment (haakonnessjoen) [inline index.js:49]: A bit weird that the password would be entered with "carriage return" \r, when all other commands are using only "line feed" \n ? Is this per spec? And why the carriage return before the password? Is this tested?

File: index.js [hunk 3]
```diff
@@ -25,61 +25,61 @@ instance.prototype.updateConfig = function(config) {
...
 	}
 
-	// Match first letter of expected response from unit.
-	else if (self.login === false && data.match(/V|60/)) {
+	// Match login sucess response from unit.
+	else if (self.login === false && data.match("Login")) {
```
Comment (haakonnessjoen) [inline index.js]: match() are usually meant to be used with regular expressions, not strings. Better to write /Login/, for clarity. (again, nitpicking, and personal feelings ;P )

**Review by haakonnessjoen — COMMENTED** [review]
Great stuff :) Want to look at these tiny things too? Maybe fix all "X" + "\n" to "X\n" too? :D

File: index.js (hunk 1 above)
Comment (EventReplay) [reply index.js]: Be nitpicky! I really appreciate it :) appeals to my OCD.

File: index.js (hunk 2 above)
Comment (EventReplay) [reply index.js:49]: I did tests on my SMX with hercules as it didn't seem to work in companion. So I could see responses \r made things work and seemed to be needed.  \r beford password guarantees it's waiting for the password with a clean line, it wasn't accepting it without it. This is same tested working code I added to the SMP module as it was requested, thought I'd have a go :) What's the worst that can happen! (DOGHOUSE)

File: index.js (hunk 3 above)
Comment (EventReplay) [reply index.js]: Understood

File: index.js (hunk 2 above)
Comment (EventReplay) [reply index.js:49]: Also setting verbose mode requires \r and needs to be the first command sent which is what gave me the idea to try it!

Comment (haakonnessjoen) [reply index.js:49]: Ok :)

**Pull request merged**